### DISCUSSION TITLE
Error while compiling

### DISCUSSION THREAD

**Discussion by Joalor64** [HaxeFlixel/flixel#3112] [Q&A]
I got this error while compiling my game

- **Haxe version:**  4.3.3
- **Flixel version:** latest
- **OpenFL version:** latest
- **Lime version:** latest
- **Affected targets:** Windows

________________________________

D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:191: characters 13-19 : Field status has different type than in flixel.addons.ui.interfaces.IFlxUIButton
D:/a/VisionSphere/VisionSphere/.haxelib/flixel-ui/2,6,0/flixel/addons/ui/interfaces/IFlxUIButton.hx:60: characters 2-38 : ... Interface field is defined here
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:1[9](https://github.com/Joalor64GH/VisionSphere/actions/runs/8710070345/job/23891129304#step:8:10)1: characters 13-19 : ... flixel.ui.FlxButtonState should be Int
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : Field set_status has different type than in flixel.addons.ui.interfaces.IFlxUIButton
D:/a/VisionSphere/VisionSphere/.haxelib/flixel-ui/2,6,0/flixel/addons/ui/interfaces/IFlxUIButton.hx:60: characters 2-38 : ... Interface field is defined here
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters [11](https://github.com/Joalor64GH/VisionSphere/actions/runs/8710070345/job/23891129304#step:8:12)-21 : ... error: flixel.ui.FlxButtonState should be Int
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... have: (Int) -> ...
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... want: (flixel.ui.FlxButtonState) -> ...
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:191: characters [13](https://github.com/Joalor64GH/VisionSphere/actions/runs/8710070345/job/23891129304#step:8:14)-19 : Field status has different type than in flixel.addons.ui.interfaces.IFlxUIButton
D:/a/VisionSphere/VisionSphere/.haxelib/flixel-ui/2,6,0/flixel/addons/ui/interfaces/IFlxUIButton.hx:60: characters 2-38 : ... Interface field is defined here
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:191: characters 13-19 : ... flixel.ui.FlxButtonState should be Int
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:6[14](https://github.com/Joalor64GH/VisionSphere/actions/runs/8710070345/job/23891129304#step:8:15): characters 11-21 : Field set_status has different type than in flixel.addons.ui.interfaces.IFlxUIButton
D:/a/VisionSphere/VisionSphere/.haxelib/flixel-ui/2,6,0/flixel/addons/ui/interfaces/IFlxUIButton.hx:60: characters 2-38 : ... Interface field is defined here
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... error: flixel.ui.FlxButtonState should be Int
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... have: (Int) -> ...
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... want: (flixel.ui.FlxButtonState) -> ...
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:[19](https://github.com/Joalor64GH/VisionSphere/actions/runs/8710070345/job/23891129304#step:8:20)1: characters 13-19 : Field status has different type than in flixel.addons.ui.interfaces.IFlxUIButton
D:/a/VisionSphere/VisionSphere/.haxelib/flixel-ui/2,6,0/flixel/addons/ui/interfaces/IFlxUIButton.hx:60: characters 2-38 : ... Interface field is defined here
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:191: characters 13-19 : ... flixel.ui.FlxButtonState should be Int
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-[21](https://github.com/Joalor64GH/VisionSphere/actions/runs/8710070345/job/23891129304#step:8:22) : Field set_status has different type than in flixel.addons.ui.interfaces.IFlxUIButton
D:/a/VisionSphere/VisionSphere/.haxelib/flixel-ui/2,6,0/flixel/addons/ui/interfaces/IFlxUIButton.hx:60: characters 2-38 : ... Interface field is defined here
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... error: flixel.ui.FlxButtonState should be Int
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... have: (Int) -> ...
D:/a/VisionSphere/VisionSphere/.haxelib/flixel/5,7,0/flixel/ui/FlxButton.hx:614: characters 11-21 : ... want: (flixel.ui.FlxButtonState) -> ...

**Comment by DetectiveBaldi** [comment]
Update `flixel-ui`
